### PR TITLE
Street systems lambda

### PR DESCRIPTION
**Pull request by LBHMVillalobos**
New file created:
lambdas/street_systems_api_ingestion/main.py
Script to call Street System API, create dataframes and write to S3

File updated: 
terraform/etl/49-lambda-streetscene-ingestion.tf
New module and trigger have been added

### CODE REVIEW

File: terraform/etl/49-lambda-streetscene-ingestion.tf [hunk 1]
```diff
@@ -120,3 +121,46 @@ resource "aws_iam_role_policy_attachment" "streetscene_street_systems_lambda_sec
...
+  lambda_output_path             = "../../lambdas/street-systems-api-ingestion.zip"
+  environment_variables          = {
+    API_SECRET_NAME       = "/data-and-insight/streets_systems_api_key"
+    OUTPUT_S3_FOLDER      = "${module.landing_zone_data_source.bucket_arn}/streetscene/*"
+    TARGET_S3_BUCKET_NAME = "streetscene/traffic-counters/"
+    API_URL = "https://flask-customer-api.ki8kabg62o4fg.eu-west-2.cs.amazonlightsail.com"
+  }
+  layers = [
+    "arn:aws:lambda:eu-west-2:120038763019:layer:requests-2-31-0-and-httplib-0-22-0-layer:1",
+    "arn:aws:lambda:eu-west-2:336392948345:layer:AWSSDKPandas-Python39:12",
```
Comment (timburke-hackit) [inline terraform/etl/49-lambda-streetscene-ingestion.tf:144]: replace account id with this interpolated value - the account numbers are different depending on which environment you're deploying to:

```suggestion
    "arn:aws:lambda:eu-west-2:${data.aws_caller_identity.data_platform.account_id}:layer:AWSSDKPandas-Python39:12",
```

**Review by timburke-hackit — CHANGES_REQUESTED** [review]

File: terraform/etl/49-lambda-streetscene-ingestion.tf (hunk 1 above)
Comment (timburke-hackit) [reply terraform/etl/49-lambda-streetscene-ingestion.tf:144]: actually, ignore this - I've mis-remembered the stg account number and thought it was this

File: terraform/etl/49-lambda-streetscene-ingestion.tf [hunk 2]
```diff
@@ -120,3 +121,46 @@ resource "aws_iam_role_policy_attachment" "streetscene_street_systems_lambda_sec
...
+  lambda_source_dir              = "../../lambdas/street-systems-api-ingestion"
+  lambda_output_path             = "../../lambdas/street-systems-api-ingestion.zip"
+  environment_variables          = {
+    API_SECRET_NAME       = "/data-and-insight/streets_systems_api_key"
+    OUTPUT_S3_FOLDER      = "${module.landing_zone_data_source.bucket_arn}/streetscene/*"
+    TARGET_S3_BUCKET_NAME = "streetscene/traffic-counters/"
+    API_URL = "https://flask-customer-api.ki8kabg62o4fg.eu-west-2.cs.amazonlightsail.com"
+  }
+  layers = [
+    "arn:aws:lambda:eu-west-2:120038763019:layer:requests-2-31-0-and-httplib-0-22-0-layer:1",
```
Comment (timburke-hackit) [inline terraform/etl/49-lambda-streetscene-ingestion.tf:143]:
```suggestion
    "arn:aws:lambda:eu-west-2:${data.aws_caller_identity.data_platform.account_id}:layer:requests-2-31-0-and-httplib-0-22-0-layer:1",
```

File: terraform/etl/49-lambda-streetscene-ingestion.tf [hunk 3]
```diff
@@ -120,3 +121,46 @@ resource "aws_iam_role_policy_attachment" "streetscene_street_systems_lambda_sec
...
+    "arn:aws:lambda:eu-west-2:336392948345:layer:AWSSDKPandas-Python39:12",
+    "arn:aws:lambda:eu-west-2:120038763019:layer:urllib3-1-26-18-layer:1",
+    "arn:aws:lambda:eu-west-2:120038763019:layer:s3fs-2023-12-2-layer:1"
```
Comment (timburke-hackit) [inline terraform/etl/49-lambda-streetscene-ingestion.tf:146]: interpolate account number for different environment accounts
```suggestion
    "arn:aws:lambda:eu-west-2:${data.aws_caller_identity.data_platform.account_id}:layer:urllib3-1-26-18-layer:1",
    "arn:aws:lambda:eu-west-2:${data.aws_caller_identity.data_platform.account_id}:layer:s3fs-2023-12-2-layer:1"
```

File: terraform/etl/49-lambda-streetscene-ingestion.tf (hunk 2 above)
Comment (timburke-hackit) [reply terraform/etl/49-lambda-streetscene-ingestion.tf:143]: this one is the stg account id though so does need to be interpolated

**Review by timburke-hackit — DISMISSED** [review]

File: terraform/etl/49-lambda-streetscene-ingestion.tf [hunk 4]
```diff
@@ -120,3 +121,46 @@ resource "aws_iam_role_policy_attachment" "streetscene_street_systems_lambda_sec
...
+  lambda_output_path             = "../../lambdas/street-systems-api-ingestion.zip"
+  environment_variables          = {
+    API_SECRET_NAME       = "/data-and-insight/streets_systems_api_key"
+    OUTPUT_S3_FOLDER      = "${module.landing_zone_data_source.bucket_arn}/streetscene/*"
+    TARGET_S3_BUCKET_NAME = "streetscene/traffic-counters/"
+    API_URL = "https://flask-customer-api.ki8kabg62o4fg.eu-west-2.cs.amazonlightsail.com"
+  }
+  layers = [
+    "arn:aws:lambda:eu-west-2:${data.aws_caller_identity.data_platform.account_id}:layer:requests-2-31-0-and-httplib-0-22-0-layer:1",
+    "arn:aws:lambda:eu-west-2:${data.aws_caller_identity.data_platform.account_id}:layer:AWSSDKPandas-Python39:12",
```
Comment (timburke-hackit) [inline terraform/etl/49-lambda-streetscene-ingestion.tf:144]: This one was OK, 336392948345 is where the AWS managed layers live and is the same regardless of where you're deploying to - sorry my fault
```suggestion
    "arn:aws:lambda:eu-west-2:336392948345:layer:AWSSDKPandas-Python39:12",
```

**Review by LBH-wgreeff — APPROVED** [review]
LGTM

**Pull request merged**